benchmarks/tensorexpr/benchmark.py

```diff
@@ -237,6 +237,61 @@ def cuda_pointwise_context(loop_levels, block_count, block_size):
     if block_size:
         torch._C._jit_set_te_cuda_pointwise_block_size(old_block_size)
 
+# Auxiliary class to facilitate dynamic input shape
+class DynamicShape(object):
+    r'''
+    An Auxiliary class for dynamic shape benchmarks
+
+    Pre-computes input with random shapes and also 
+    modifies the compute method so in each call the
+    fuser sees a different input tensor shape
+    '''
+
+    # number of random inputs in an instance
+    SAMPLE_SIZE = 100
+
+    def __init__(self, dynamic_range=1.2):
+        self._input_samples = []
+        self._input_sample_index = 0
+        self._dynamic_range = 1. / dynamic_range if dynamic_range > 1.0 else dynamic_range
+
+    # returns the input test case that current index points to
+    @property
+    def inputs(self):
+        return self._input_samples[self._input_sample_index]
+
+    # an inputs assignment actually adds a test case in the class buffer
+    @inputs.setter
+    def inputs(self, val):
+        self._input_samples.append(val)
+
+    # runs normal compute while increment test case index
+    def compute(self):
+        super().compute()
+        self._input_sample_index = (self._input_sample_index + 1) % self.SAMPLE_SIZE
+
+    # defined by benchmark, the benchmark needs to specify the input
+    # tensor construction in this method, essentially the same way
+    # a benchmark creates the inputs list in the initializer
+    def instantiate_input(self):
+        raise NotImplementedError
+
+    # pre-compute inputs so the creations of random tensors
+    # do not add to the compute time
+    def load_inputs(self):
+        for i in range(self.SAMPLE_SIZE - 1):
+            self.instantiate_input()
+
+    # returns a randomized shape
+    def rand_shape(self, shape):
+        if 'PYTORCH_CUDA_FUSER_NO_DYNSHAPE' in os.environ:
+            return shape
```

Comment:
It would be cool if this were a command line argument instead of an environment variable since env isn't very discoverable.  But, meh, it's a benchmark harness.  Maybe in a follow up diff if you feel compelled :-).

Reply:
Seems like a reasonable request to me.

Reply:
@bertmaher Thanks for the suggestion. I just modified the class to take this option from command line.

Reply:
Thanks for the quick update, will merge now!